DiscordBot/Program.cs

```diff
@@ -40,9 +47,83 @@ private static async Task MainAsync()
             commands.RegisterCommands<EventCommands>();
 
             discord.UseInteractivity(new InteractivityConfiguration());
+            discord.MessageReactionAdded += async (client, eventArguments) =>
+            {
+                await OnMessageReactionAdded(client, eventArguments, eventsSheetsService);
+            };
 
             await discord.ConnectAsync();
             await Task.Delay(-1);
         }
+
+        private static async Task OnMessageReactionAdded(
+            DiscordClient client,
+            MessageReactionAddEventArgs eventArguments,
+            IEventsSheetsService eventsSheetsService)
+        {
+            if (eventArguments.User == client.CurrentUser)
+            {
+                return;
+            }
+
+            var discordEvent = await GetEventFromMessageIdOrDefaultAsync(eventArguments.Message.Id, eventsSheetsService);
+            if (discordEvent == null)
+            {
+                return;
+            }
+
+            await ProcessReaction(client, eventArguments, discordEvent, eventsSheetsService);
+
+            await UpdateSignupSheet(eventArguments, discordEvent, eventsSheetsService);
+        }
+
+        private static async Task ProcessReaction(
+            DiscordClient client,
+            MessageReactionAddEventArgs eventArguments,
+            DiscordEvent discordEvent,
+            IEventsSheetsService eventsSheetsService)
+        {
+            var dmChannel = await eventArguments.Guild
+                .GetMemberAsync(eventArguments.User.Id).Result
+                .CreateDmChannelAsync();
+
+            if (eventArguments.Emoji.GetDiscordName() == ":no_entry_sign:")
+            {
+                await eventsSheetsService.ClearResponsesForUserAsync(discordEvent.Key, eventArguments.User.Id);
+                await client.SendMessageAsync(dmChannel, $"You've signed off {discordEvent.Name}.");
+            }
+            else
+            {
+                try
+                {
+                    await eventsSheetsService.AddResponseForUserAsync(
+                        discordEvent.Key,
+                        eventArguments.User.Id,
+                        eventArguments.Emoji.GetDiscordName());
+                }
+                catch (ResponseNotFoundException)
+                {
+                    return;
+                }
+
+                await client.SendMessageAsync(
+                    dmChannel,
+                    $"You've responded to {discordEvent.Name} as {eventArguments.Emoji.Name}.");
+            }
+        }
+
+        private static async Task UpdateSignupSheet(
```

Comment:
Is this actually updating the sign up sheet?

Reply:
It refers to the sheet on the signup message, I'll rename it to UpdateSignupMessage to make it more clear.